pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -164,11 +165,12 @@ func buildVirtualHostDomains(service *model.Service, port int, node model.Proxy)
 	domains = append(domains, generateAltVirtualHosts(service.Hostname, port, node.Domain)...)
 
 	if len(service.Address) > 0 {
+		svcAddr := service.GetServiceAddressForProxy(&node)
 		// add a vhost match for the IP (if its non CIDR)
-		cidr := util.ConvertAddressToCidr(service.Address)
+		cidr := util.ConvertAddressToCidr(svcAddr)
```

Comment:
It may be ok for now (and for the case kubedns is used) - but not sure if it's the best option once we have coreDNS or 
other non-kubedns options. But we can start with this, and later add the full list of IPs.

Can you just add a TODO and comment indicating the options and why ?

Reply:
I agree. Out of the full list of IPs available for an endpoint, we should add support to indicate what IPs are exposed, and what IPs should be used as service IPs. 

will add TODO and comments in next PR.